testing/SetupTestbed.ps1

```diff
@@ -366,13 +380,63 @@ it is likely that the DNS entry was still successfully added. To
 verify, log on to DC1 and run 'Resolve-DnsName ls1' in PowerShell.
 If it returns NXDOMAIN, you'll need to add it manually."
 Write-Output "The time is $(Get-Date)."
-az vm run-command create `
+
+# Define the PowerShell script with the DomainName variable interpolated
+$scriptContent = @"
+`$scriptBlock = {
+    Add-DnsServerResourceRecordA -Name LS1 -ZoneName $DomainName -AllowUpdateAny -IPv4Address $LsIP -TimeToLive 01:00:00 -AsJob

```

Comment:
@cbaxley one other note, it's possible that the problem we're running into is because the zone we're telling Windows to add isn't "fully qualified", as in all the way to the root zone, `.`. I tried to test this, but it's really hard to say conclusively with these "sometimes" bugs.

- Testing with the dot after the domain name: successful 5 out of 5 times.
- Testing without the dot after the domain name: successful 3 out of 5 times.

So...maybe this fixes it?

```suggestion
    Add-DnsServerResourceRecordA -Name LS1 -ZoneName $DomainName. -AllowUpdateAny -IPv4Address $LsIP -TimeToLive 01:00:00 -AsJob

```

Reply:
Done